cppcache/src/PdxRemoteWriterFactoryImpl.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PdxRemoteWriterFactoryImpl.hpp"
+
+#include "PdxRemoteWriter.hpp"
+#include "PdxType.hpp"
+#include "PdxTypeRegistry.hpp"
+#include "util/cxx_extensions.hpp"
+
+namespace apache {
+namespace geode {
+namespace client {
+
+PdxRemoteWriterFactoryImpl::~PdxRemoteWriterFactoryImpl() = default;
+
+std::unique_ptr<PdxRemoteWriter> PdxRemoteWriterFactoryImpl::create(
```

Comment:
It really seems like you could test this at unit level by mocking `PdxTypeRegistery` and `PdxType`. Make this a member function back from where it came and test it a the unit level. This avoids the creation of this factory, the exporting of internal symbols, and the interdiction of artificial behavior into the production implementation, via overriding this implementation.

Reply:
Your approach seems promising. I'll look into it and see if it's feasible.